File: python/88_Merge_Sorted_Array.py
class Solution:
    # @param {integer[]} nums1
    # @param {integer} m
    # @param {integer[]} nums2
    # @param {integer} n
    # @return {void} Do not return anything, modify nums1 in-place instead.
    def merge(self, nums1, m, nums2, n):
        if m==0:
            i=0
            while i<n:
                nums1[i]=nums2[i]
                i+=1
        elif n>0:
            cur,n1,n2=m+n-1,m-1,n-1
            while n1>=0 and n2>=0:
                if nums1[n1]>nums2[n2]:
                    nums1[cur]=nums1[n1]
                    n1-=1
                else:
                    nums1[cur]=nums2[n2]
                    n2-=1
                cur-=1
            while n2>=0:
                nums1[n2]=nums2[n2]
                n2-=1

File: Java/88_Merge_Sorted_Array.java
public class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i;
        if(n<=0)return ;
        else if(m<=0){
            for(i=0;i<n;i++){
                nums1[i]=nums2[i];
            }
            return;
        }
        i=m+n-1;
        while(m>0&&n>0){
            if(nums1[m-1]>nums2[n-1]){
                nums1[i]=nums1[m-1];
                m--;
            }else{
                nums1[i]=nums2[n-1];
                n--;
            }
            i--;
        }
        while(n>0){
            nums1[i]=nums2[n-1];
            i--;
            n--;
        }
    }
}